pkg/polaris/aws/exocompute_test.go

```diff
@@ -0,0 +1,177 @@
+//   - AWS_SHARED_CREDENTIALS_FILE=<path-to-aws-credentials-file>
+//   - AWS_CONFIG_FILE=<path-to-aws-config-file>
+//
+// The file referred to by TEST_AWSACCOUNT_FILE should contain a single
+// testAwsAccount JSON object.
+func TestAwsExocompute(t *testing.T) {
+	ctx := context.Background()
+
+	if !testsetup.BoolEnvSet("TEST_INTEGRATION") {
+		t.Skipf("skipping due to env TEST_INTEGRATION not set")
+	}
+
+	testAccount, err := testsetup.AWSAccount()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	awsClient := Wrap(client)
+
+	// Adds the AWS account identified by the specified profile to RSC. Note
+	// that the profile needs to have a default region.
+	accountID, err := awsClient.AddAccount(ctx, Profile(testAccount.Profile), core.FeatureCloudNativeProtection,
+		Name(testAccount.AccountName), Regions("us-east-2"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Enable the exocompute feature for the account.
+	exoAccountID, err := awsClient.AddAccount(ctx, Profile(testAccount.Profile), core.FeatureExocompute,
+		Regions("us-east-2"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if accountID != exoAccountID {
+		t.Fatalf("cloud native protection and exocompute added to different cloud accounts: %q vs %q",
+			accountID, exoAccountID)
+	}
+
+	// Verify that the account was successfully added.
+	account, err := awsClient.Account(ctx, CloudAccountID(accountID), core.FeatureExocompute)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if account.Name != testAccount.AccountName {
+		t.Errorf("invalid name: %v", account.Name)
+	}
+	if account.NativeID != testAccount.AccountID {
+		t.Errorf("invalid native id: %v", account.NativeID)
+	}
+	if n := len(account.Features); n == 1 {
+		if account.Features[0].Name != "EXOCOMPUTE" {
+			t.Errorf("invalid feature name: %v", account.Features[0].Name)
+		}
+		if regions := account.Features[0].Regions; !reflect.DeepEqual(regions, []string{"us-east-2"}) {
+			t.Errorf("invalid feature regions: %v", regions)
+		}
+		if account.Features[0].Status != "CONNECTED" {
+			t.Errorf("invalid feature status: %v", account.Features[0].Status)
+		}
+	} else {
+		t.Errorf("invalid number of features: %v", n)
+	}
+
+	exoID, err := awsClient.AddExocomputeConfig(ctx, AccountID(testAccount.AccountID),
+		Managed("us-east-2", testAccount.Exocompute.VPCID,
+			[]string{testAccount.Exocompute.Subnets[0].ID, testAccount.Exocompute.Subnets[1].ID}))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Retrieve the exocompute config added.
+	exoConfig, err := awsClient.ExocomputeConfig(ctx, exoID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if exoConfig.ID != exoID {
+		t.Errorf("invalid id: %v", exoConfig.ID)
+	}
+	if exoConfig.Region != "us-east-2" {
+		t.Errorf("invalid region: %v", exoConfig.Region)
+	}
+	if exoConfig.VPCID != testAccount.Exocompute.VPCID {
+		t.Errorf("invalid vpc id: %v", exoConfig.VPCID)
+	}
+	if exoConfig.Subnets[0].ID != testAccount.Exocompute.Subnets[0].ID && exoConfig.Subnets[0].ID != testAccount.Exocompute.Subnets[1].ID {
```

Comment:
nits: I initially thought this was a copy/paste bug :). Maybe add a comment to clarify that we checks that the retrieved config must match one of the test accounts subnets or something? Or refactor a bit, but that might feel overkill, e.g.:

```
s1 := testAccount.Exocompute.Subnets[0]
s2 := testAccount.Exocompute.Subnets[1]

if got := exoConfig.Subnets[0]: got.ID != s1.ID && got.ID != s2.ID {
    // ...
}
if got := exoConfig.Subnets[1]; got.ID != s1.ID && got.ID != s2.ID {
    // ...
}
```